capa/features/extractors/vmray/models.py

```diff
@@ -276,7 +276,7 @@ class ElfFileHeader(BaseModel):
 
 class ElfFile(BaseModel):
     # file_header: ElfFileHeader
-    sections: Optional[list[ElfFileSection]] = None
+    sections: list[ElfFileSection] = []
```

Comment:
incidentally, is this the correct way to set the default value, particularly as a list? i see this pattern used throughout the file.

my worry is that the default value `= []` uses the same instance of a mutable list, rather than copies of it. sorta like when you have a kwarg parameter `def foo(bar=[])`. 

in the past, i've used `pydantic.Field` for these. but maybe pydantic is extra smart and doesn't require this. @mr-tz @mike-hunhoff

Reply:
hm, great question, this is how mypy accepted the change and I saw the pattern throughout. Other files use `Optional[list[<foo>]] = None` or Field, we should cleanup the inconsistencies (separately).

Reply:
My understanding is that pydantic handles this correctly (i.e. deep copy) for non-hashable default values (i.e. lists). source: https://docs.pydantic.dev/latest/concepts/models/#fields-with-non-hashable-default-values